pkg/pool-manager/pool.go

```diff
@@ -176,3 +177,24 @@ func getNextMac(currentMac net.HardwareAddr) net.HardwareAddr {
 
 	return currentMac
 }
+
+// Checks if the namespace of an instance is opted in for kubemacpool
+func (p *PoolManager) IsInstanceOptedIn(namespaceName string, labelSelector string) (error, bool) {
+
+	isNamespaceOptedIn := false
+	ns, err := p.kubeClient.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+
+	if err != nil {
+		return err, isNamespaceOptedIn
+	}
+
+	log.V(1).Info("namespaceName Labels", "Labels", ns.GetLabels())
+	for label := range ns.GetLabels() {
+		if label == labelSelector {
+			isNamespaceOptedIn = true
```

Comment:
Would you please check the label value as well. So we make sure that users are following allowed values. Otherwise it can happen that they will set the value to "FOO". Then, once we fix this and check for a specific value, they would be surprised KMP does not work anymore.